external/kafka/pom.xml

```diff
@@ -50,8 +62,35 @@
     </dependency>
     <dependency>
       <groupId>org.apache.kafka</groupId>
-      <artifactId>kafka_${scala.binary.version}</artifactId>
-      <version>0.8.2.1</version>
+      <artifactId>kafka_2.10</artifactId>
+      <version>0.9.0.0</version>
```

Comment:
I don't think you can update to 0.9.x without breaking a lot of compatibility. This kind of change would have to wait for Spark 2.x.